sdk/search/Azure.Search/src/Models/SuggestResults{T}.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.Core;
+
+#pragma warning disable SA1402 // File may only contain a single type
+
+namespace Azure.Search.Models
+{
+    // Hide the untyped SuggestDocumentsResult
+    [CodeGenSchema("SuggestDocumentsResult")]
+    internal partial class SuggestDocumentsResult { }
+
+    /// <summary>
+    /// Response containing suggestion query results from an index.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The .NET type that maps to the index schema. Instances of this type can
+    /// be retrieved as documents from the index.
+    /// </typeparam>
+    [JsonConverter(typeof(SuggestResultsConverterFactory))]
+    public partial class SuggestResults<T>
+    {
+        /// <summary>
+        /// A value indicating the percentage of the index that was included in
+        /// the query, or null if minimumCoverage was not set in the request.
+        /// </summary>
+        public double? Coverage { get; internal set; }
+
+        /// <summary>
+        /// The sequence of suggestions returned by the query.
+        /// </summary>
+        public IList<SearchSuggestion<T>> Results { get; internal set; } =
```

Comment:
Should this be mutable?

Reply:
Ditto on the above for all of these return model types - I want to track the generator for the time being so it doesn't feel like two separate API experiences in one client library.